src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,72 @@
+.calendar {
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: containerWidth(
+    $cell-width,
+    $cells-gap,
+    $cells-per-row,
+    $cells-container-padding
+  );
+
+  @each $dayname, $daynumber in $days-names {
+    &--start-day--#{$dayname} &__day:first-child {
+      margin-left: marginForFirstWeekDay($daynumber);
+    }
+  }
+
+  @each $dayname, $daynumber in $days-names {
+    &--start-day--#{$dayname}
+    &__day:nth-child(7n + #{7 - $daynumber + 1}) {
+      background-color: #ddd;
+    }
+  }
+
+  @each $dayname, $daynumber in $days-names {
+    &--start-day--#{$dayname}
+    &__day:nth-child(7n + #{7 - $daynumber + 1}):hover {
+      background-color: pink;
+    }
+  }
+
+  @each $dayname, $daynumber in $days-names {
+    &--start-day--#{$dayname}
+    &__day:nth-child(7n + #{7 - $daynumber + 1})::before {
+      color: maroon;
+    }
+  }
```

Comment:
you have used the same loop 4 times. You can use only once. Just use different selectors inside loop body

example: 
```
  @each {{your_condition}} {
    &__{{your_first_selector}} {
     {{your_styles}} 
    }
    &__{{your_second_selector}} {
     {{your_styles}} 
    }
  }
```